### DISCUSSION TITLE
promise

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#74] [프론트 기술 질문 공간]
promise 의 내장 메서드들의 종류와 사용법을  알려주세요

**Comment by javascriptbiggosoo** [comment]
Promise.all(iterable)
주어진 이터러블(iterable)의 모든 Promise 객체가 성공적으로 해결될 때까지 대기한 후, 모든 Promise의 결과 값을 담은 Promise 객체를 반환합니다.
iterable: Promise 객체로 이루어진 이터러블(iterable).
반환된 Promise 객체는 모든 Promise 객체가 성공적으로 해결될 때 해결됩니다. 그렇지 않으면 가장 먼저 거부된 Promise의 이유로 거부됩니다.

Promise.race(iterable)
주어진 이터러블(iterable)의 Promise 객체 중 가장 먼저 성공 또는 거부된 Promise 객체의 결과 값을 담은 Promise 객체를 반환합니다.
iterable: Promise 객체로 이루어진 이터러블(iterable).
반환된 Promise 객체는 가장 먼저 성공 또는 거부된 Promise의 결과 값으로 처리됩니다.

**Comment by michoball** [comment]
promise는 총 4개의 정적 메서드를 가지고 있습니다. 
promise.all() :  all 안의 promise 배열들 병렬적으로 실행하며 전체가 resolve 되어야 resolve되고 하나라도 reject 되면 reject됩니다. 

promise.allSettled():  allSettled 안의 promise 배열들 병렬적으로 실행하며 실행이 완료되면 완료된 reject나 resolve 값을 객체 배로 반환합니다. 

promise.any(): all과 반대로  하나라도 fulfilled 되면 해당 값을 반환하고 전부다 rejected 되면 reject를 반환합니다. 

promise.race(): race안 promise를 실행해 제일 먼저 fulfilled된 값 반환합니다.
